### PR TITLE
Adjust devcontainer

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

- Remove pacakges from dockerfile that is in the base image
- Pin `sassc` to `2.1.0` due to slowdown issues with later versions (162198ms vs 35700ms total build time) 
- Move `DEVCONTAINER` env to devcontainer.json
- Switch devcontainer terminal to zsh

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: 

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by frenck** [comment]
The context is missing, see: https://github.com/home-assistant/home-assistant.io/pull/16408

File: Dockerfile [hunk 1]
```diff
@@ -20,9 +17,6 @@ ENV \
 RUN apt update && export DEBIAN_FRONTEND=noninteractive \
     && apt-get install -y --no-install-recommends \
         ack \
-        git \
-        locales \
-        procps \
     && echo "en_US UTF-8" > /etc/locale.gen \
     && locale-gen en_US.UTF-8 \
     && echo 'export PS1="\\w\$ "' > /root/.bashrc \
```
Comment (balloob) [inline Dockerfile]: Is this still necessary?

Comment (frenck) [reply Dockerfile]: No, I guess we can just switch to the same as the core devcontainer and push zsh in 😉

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-ARG VARIANT=2.6
+ARG VARIANT=2.7
```
Comment (frenck) [inline Dockerfile]: The ruby version pinned in .ruby-version is 2.6.2 (and the one used and cached by Netlify).
This change makes the versions out of sync.

That said, Netlify's build image seems to carry 2.7.1 as well (as main), so adjusting the version file to `2.7.1` will do the trick. (Although I'm not 100% sure how that impacts the used bundler version).

Comment (ludeeus) [reply Dockerfile]: I'll revert this

File: Dockerfile [hunk 3]
```diff
@@ -7,28 +7,23 @@ RUN \
...
-    && locale-gen en_US.UTF-8 \
-    && echo 'export PS1="\\w\$ "' > /root/.bashrc \
-    && apt-get autoremove -y \
-    && apt-get clean -y \
-    && rm -rf /var/lib/apt/lists/*
+# Install tools
+RUN \
+  apt update \
+  && DEBIAN_FRONTEND=noninteractive \
+  && apt-get install -y --no-install-recommends \
```
Comment (ludeeus) [inline Dockerfile]:
```suggestion
  && DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
```

**Review by frenck — APPROVED** [review]
Tested in codespaces 👍

**Pull request merged**